website/pages/category/[id].tsx

```diff
@@ -8,78 +8,84 @@ import { CategoryData, RestaurantData } from "@/types/category-by-id";
-          .from("categories")
-          .select("*")
-          .eq("id", id)
-          .single();
-
-        if (error) throw error;
-
-        setCategoryData(data);
-
-        const { data: categoriesData, error: categoriesError } = await supabase
-          .from("categories")
-          .select("id, restaurant_id, image")
-          .neq("restaurant_id", 0)
-          .contains("tags", [data.name.toLowerCase()]);
-
-        if (categoriesError) throw categoriesError;
-
-        const restaurant = await Promise.all(
-          categoriesData.map(async (category) => {
-            const { data: restaurantData, error: restaurantError } =
-              await supabase
-                .from("restaurants")
-                .select("id, name, address")
-                .eq("id", category.restaurant_id)
-                .single();
-
-            if (restaurantError) throw restaurantError;
-
-            return {
-              ...restaurantData,
-              image: category.image,
-              rating: 0,
-              reviews: 0,
-            };
-          })
-        );
-
-        setRestaurantsData(restaurant);
-        setIsRestaurantsLoading(false);
-      } catch (error) {
-        console.error("Error fetching category data:", error);
+      if (suffix) {
+        try {
+          const { data, error } = await supabase
+            .from("categories")
+            .select("*")
+            .eq("id", atob(suffix!))
+            .single();
+
+          if (error) throw error;
+
+          setCategoryData(data);
+
+          const { data: categoriesData, error: categoriesError } =
+            await supabase
+              .from("categories")
+              .select("id, restaurant_id, image")
+              .neq("restaurant_id", 0)
+              .contains("tags", [data.name.toLowerCase()]);
+
+          if (categoriesError) throw categoriesError;
+
+          const restaurant = await Promise.all(
+            categoriesData.map(async (category) => {
+              const { data: restaurantData, error: restaurantError } =
+                await supabase
+                  .from("restaurants")
+                  .select("id, name, address")
+                  .eq("id", category.restaurant_id)
+                  .single();
+
+              if (restaurantError) throw restaurantError;
+
+              return {
+                ...restaurantData,
+                image: category.image,
+                rating: 0,
+                reviews: 0,
+              };
+            })
+          );
+
+          setRestaurantsData(restaurant);
+        } catch (error) {
+          console.error("Error fetching category data:", error);
+        } finally {
+          setIsRestaurantsLoading(false);
+        }
       }
     };
 
     fetchCategoryData();
-  }, [id]);
+  }, [id, router, suffix]);
 
   return (
     <RootLayout>
       <section className="py-16 md:py-20 lg:py-28">
-        {!isRestaurantsLoading && categoryData ? (
+        {categoryData ? (
```

Comment:
I think `isRestaurantsLoading` should be checked as well, until category loads need to show loading placeholder.

Reply:
- `isRestaurantsLoading` is different than category loading.
- And currently loader is showing when everything is loading.
- 
[Screencast from 05-04-24 03:32:36 PM IST.webm](https://github.com/canopas/bite-space/assets/123158106/0afb5910-d663-4af7-ad38-d6de07206d65)